oak_crypto/src/encryptor.rs

```diff
@@ -32,6 +32,7 @@ use async_trait::async_trait;
 /// Info string used by Hybrid Public Key Encryption;
 pub(crate) const OAK_HPKE_INFO: &[u8] = b"Oak Hybrid Public Key Encryption v1";
 
+#[derive(Clone)]
```

Comment:
We should not be cloning key pairs. I assume this is only needed to make the group key the same as the instance key. In that case I would prefer a separate special function for doing that rather than just deriving `Clone`.  This seems to indicate that cloning keys are fine.

Reply:
Removed the `Clone`, and make the keys `Arc` for now (with a TODO)

But we still need to get access to the private key for 2 reasons:
- We need to send private keys between the layers (removed once we add RKernel Orchestrator)
- We need to encrypt the private key and send in to another enclave (this could be implemented in the future as part of the `oak_crypto` crate, though it would mean that we are essentially implementing Tink)

For now I added a `get_private_key()` method to the `EncryptionKeyProvider` to solve these issues. But updating it would require further discussion (i.e. how do we want to protect private keys in Oak in general).